lib/scout_apm/tracked_request.rb

```diff
@@ -267,6 +271,14 @@ def record!
       # Bail out early if the user asked us to ignore this uri
       return if @agent_context.ignored_uris.ignore?(annotations[:uri])
 
+      # Apply name override if needed
+      unless name_override.nil?
+        scope_layer = layer_finder.scope
+        unless scope_layer.nil?
+          scope_layer.name = name_override
+        end
+      end
```

Comment:
@cschneid As I was poking around a bit more, I found this `scope` layer concept. Is this a reasonable way to achieve what we want here? Or does the rename need to be a `LayerConverter`?

Reply:
`if name_override` and `if scope_layer` - prefer positive conditions.

This is right though, pull it out into a method like `apply_name_override()` or something?